Unknown exchange coinbase in packaged version if already had coinbase registered

## Problem Definition

The last 2 packaged release versions show coinbase as unknown exchange if there was already a key in the DB.

```
03/01/2020 18:11:12 CET -- DEBUG:rotkehlchen.premium.sync:sync from server stopped -- local DB more recent than remote                                                                                                                                                           
03/01/2020 18:11:12 CET -- INFO:rotkehlchen.exchanges.exchange:Initialized kraken exchange                                                                                                                                                                                       
03/01/2020 18:11:12 CET -- INFO:rotkehlchen.exchanges.exchange:Initialized poloniex exchange                                                                                                                                                                                     
03/01/2020 18:11:12 CET -- INFO:rotkehlchen.exchanges.exchange:Initialized bittrex exchange                                                                                                                                                                                      
03/01/2020 18:11:12 CET -- INFO:rotkehlchen.exchanges.exchange:Initialized binance exchange                                                                                                                                                                                      
03/01/2020 18:11:12 CET -- CRITICAL:root:Traceback (most recent call last):                                                                                                                                                                                                      
  File "rotkehlchen/exchanges/manager.py", line 82, in initialize_exchanges                                                                                                                                                                                                      
  File "importlib/__init__.py", line 127, in import_module                                                                                                                                                                                                                       
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import                                                                                                                                                                                                                
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load                                                                                                                                                                                                              
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked                                                                                                                                                                                                     
ModuleNotFoundError: No module named 'rotkehlchen.exchanges.coinbase'                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                 
During handling of the above exception, another exception occurred:                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                 
Traceback (most recent call last):                                                                                                                                                                                                                                               
  File "rotkehlchen/server.py", line 520, in unlock_user                                                                                                                                                                                                                         
  File "rotkehlchen/rotkehlchen.py", line 188, in unlock_user                                                                                                                                                                                                                    
  File "rotkehlchen/exchanges/manager.py", line 86, in initialize_exchanges                                                                                                                                                                                                      
AssertionError: Tried to initialize unknown exchange coinbase. Should never happen.      
```

## Task

Fix this problem.

Probably did not affect anyone but still needs to be fixed in a patch release.

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -2,6 +2,8 @@
 Changelog
 =========
 
+* :bug:`602` A bug that lead to the coinbase exchange being unusable in last 2 releases was fixed.
+
 * :release:`1.0.6 <2019-12-31>`
 * :bug:`589` If there is an error an unexpected error during sign-in properly catch it and add a log entry.
 * :bug:`588` The electron log is now written in the proper directory depending on the Operating system.

--- a/rotkehlchen.spec
+++ b/rotkehlchen.spec
@@ -5,6 +5,7 @@ import platform
 import sys
 from distutils.spawn import find_executable
 
+from rotkehlchen.exchanges.manager import SUPPORTED_EXCHANGES
 from rotkehlchen.utils.misc import get_system_spec
 
 """
@@ -70,12 +71,19 @@ executable_name = 'rotkehlchen-{}-{}'.format(
     get_system_spec()['rotkehlchen'],
     'macos' if platform.system() == 'Darwin' else platform.system().lower())
 
+hiddenimports = ['cytoolz.utils', 'cytoolz._signatures']
+# Since the exchanges are loaded dynamically and some of them may not be detected
+# by pyinstaller (https://github.com/rotki/rotki/issues/602) make sure they are
+# all included as imports in the created executable
+for exchange_name in SUPPORTED_EXCHANGES:
+    hiddenimports.append(f'rotkehlchen.exchanges.{exchange_name}')
+
 a = Entrypoint(
     'rotkehlchen',
     'console_scripts',
     'rotkehlchen',
     hookspath=['tools/pyinstaller_hooks'],
-    hiddenimports=['cytoolz.utils', 'cytoolz._signatures'],
+    hiddenimports=hiddenimports,
     datas=[
         ('rotkehlchen/data/token_abi.json', 'rotkehlchen/data'),
         ('rotkehlchen/data/all_assets.json', 'rotkehlchen/data'),

